File: frontend/src/types/article.ts
export type ArticleType = 'OVERSEAS' | 'DOMESTIC' | 'CITY' | 'COUNTRYSIDE';

export interface ArticleCatalogVO {
    id: number;
    title: string;
    description: string;
    coverImg: string;
    articleType: ArticleType;
    viewCount: number;
    likeCount: number;
}


export interface ArticleDetailVO {
    id: number;
    title: string;
    content: string;
    articleType: ArticleType;
    viewCount: number;
    likeCount: number;
    createTime: string; 
}

File: backend/src/main/java/com/example/demo/entity/Article.java
package com.example.demo.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.example.demo.common.enums.ArticleType;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("article")
public class Article {

    @TableId(type = IdType.AUTO)
    private Long id;

    @Schema(description = "文章标题")
    private String title;

    @Schema(description = "文章内容")
    private String content;

    @Schema(description = "文章描述")
    private String description;

    @Schema(description = "文章封面图")
    private String coverImg;

    @Schema(description = "文章类型")
    private ArticleType articleType;

    @Schema(description = "浏览数")
    private Long viewCount;

    @Schema(description = "点赞数")
    private Long likeCount;

    @Schema(description = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @Schema(description = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @Version
    private Integer version;
}
